mesonbuild/modules/python.py

```diff
@@ -289,8 +316,9 @@ def links_against_libpython():
     return bool(cmd.get_libraries(Extension('dummy', [])))
 
 print(json.dumps({
+  'base_prefix': sys.base_prefix,
   'variables': sysconfig.get_config_vars(),
-  'paths': sysconfig.get_paths(),
+  'paths': paths,
   'install_paths': install_paths,
```

Comment:
install_paths is not defined any more, and base_prefix seems unused. Other than that, it seems to give the right values.

I am not an expert in python echosystem, but I thought distutils was about to be deprecated. Maybe that should be used only as fallback if sysconfig gives a value that is not in sys_paths ?

Reply:
The problem is that sysconfig was never used to install packages and distros only patched distutils. This was recently uncovered/rectified by pip which compared both at runtime and warned the user to file a bug with their distro.

So the safest way to handle this would be to use distutils and fall back to sysconfig.

Reply:
> install_paths is not defined any more, and base_prefix seems unused. Other than that, it seems to give the right values.

Sorry for the bad cleanup, I will fix this later.

> I am not an expert in python echosystem, but I thought distutils was about to be deprecated. Maybe that should be used only as fallback if sysconfig gives a value that is not in sys_paths ?

It is, I didn't keep that logic here to make it easier for you to verify the disutils bit.

> The problem is that sysconfig was never used to install packages and distros only patched distutils. This was recently uncovered/rectified by pip which compared both at runtime and warned the user to file a bug with their distro (but that doesn't fix older distros)

I think by distros you mean Debian, and Debian-based, which just inherit the package. Fedora for eg. currently patches sysconfig. Recent-ish pip versions should use sysconfig to install IIRC.

> So the safest way to handle this would be to use distutils and fall back to sysconfig once distutils is really gone (not much visible upstream movement there atm...)

IMO we should use sysconfig on Python >= 3.12, and distutils on older if it is patched by Debian.

I think Debian will drop their patches in favor of https://www.python.org/dev/peps/pep-0668/ in 3.12 but I am not sure, and we won't know until they release it. But considering that there won't be distutils anymore, if they keep the patches, they will probably move them to sysconfig.